cnf-certification-test/lifecycle/suite.go

```diff
@@ -448,6 +448,24 @@ func testPodsRecreation(env *provider.TestEnvironment) { //nolint:funlen
 		tnf.ClaimFilePrintf("%s", claimsLog.GetLogLines())
 		ginkgo.Fail("Some deployments or stateful sets are not in a good initial state. Cannot perform test.")
 	}
+
+	// Filter out pods with Node Assignments present and FAIL them.
+	// We run into problems with this test when there are nodeSelectors assigned affecting where
+	// pods are scheduled.  Also, they are not allowed in general, see the node-selector test case.
+	// Skip the safeguard for any pods that are using a runtimeClassName.  This is potentially
+	// because pods that are derived from a performance profile might have a built-in nodeSelector.
+	var podsWithNodeAssignement []*provider.Pod
+	for _, put := range env.Pods {
+		if !put.IsRuntimeClassNameSpecified() && put.HasNodeSelector() {
```

Comment:
@ramperher See this proposed change.  I only error out early if there are no runtimeClasses specified.  Does that satisfy some of the issues we are seeing?

Reply:
It makes sense for me, let's try it on DCI! (I'll remove the dependency with my change, so that we can test the current deployment we're doing)

Reply:
It makes sense for me, let's try it on DCI! (I'll remove the dependency with my change, so that we can test the current deployment we're doing)